src/main/java/org/hibernate/infra/replicate/jira/service/jira/handler/JiraIssueAbstractEventHandler.java

```diff
@@ -213,8 +213,18 @@ private JiraUser toUser(String value) {
 		return new JiraUser(context.projectGroup().users().mappedPropertyName(), value);
 	}
 
-	private Optional<JiraTransition> prepareTransition(JiraIssue sourceIssue) {
-		return statusToTransition(sourceIssue.fields.status.id).map(JiraTransition::new);
+	protected Optional<JiraTransition> prepareTransition(JiraSimpleObject sourceStatus, JiraIssue destIssue) {
+		String downstreamStatus = context.projectGroup().statuses().mapping()
+				.get(sourceStatus.name.toLowerCase(Locale.ROOT));
+		return prepareTransition(downstreamStatus, destIssue);
+	}
+
+	protected Optional<JiraTransition> prepareTransition(String downstreamStatus, JiraIssue destIssue) {
+		return Optional
+				.ofNullable(statusToTransition(destIssue.fields.status.name, downstreamStatus,
+						() -> findRequiredTransitionId(downstreamStatus, destIssue))
+						.orElseGet(() -> findRequiredTransitionId(downstreamStatus, destIssue).orElse(null)))
```

Comment:
This duplicate `findRequiredTransitionId` call is suspicious... Can `statusToTransition` actually return an empty optional here? If it does, doesn't it mean `findRequiredTransitionId` returned an empty optional and thus isn't useful?

Reply:
Yeah, that was a bad choice 🙈 
The thing is that the transition is cached, but that first attempt to call `findRequiredTransitionId` may end up failing for some reason (since it's a rest call), so that was a silly attempt to still try and get a transition if it cached the failed attempt (`null`) *BUT* why would one cache the `null` in the first place!? 🤦🏻‍♂️ 
It should be better now with nulls not cached.